logstash-core/lib/logstash/util/substitution_variables.rb

```diff
@@ -76,6 +79,19 @@ def replace_placeholders(value)
       end
       replacement.to_s
     end
+
+    # no further action need if substitution didn't happen
+    return placeholder_value unless is_placeholder_found
+
+    # ENV ${var} value may carry single quote or escaped double quote
+    # or single/double quoted entries in array string, needs to be refined
+    refined_value = placeholder_value.gsub(/[\\"\\']/, '')
```

Comment:
I'd suggest trimming whitespace before and after the brackets, so that `start_with` and `end_with` can work with strings like `' ["a", "b", "c"] '`